src/batch/src/executor/join/lookup_join.rs

```diff
@@ -62,6 +64,31 @@ impl<C: BatchTaskContext> ProbeSideSource<C> {
     /// Creates the `RowSeqScanNode` that will be used for scanning the probe side table
     /// based on the passed `RowRef`.
     fn create_row_seq_scan_node(&self, cur_row: &RowRef) -> Result<NodeBody> {
+        // Check that the data types of both sides of the equality predicate are the same
+        // TODO: Handle the cases where the data types of both sides are different but castable
+        // (e.g. int32 and int64)
+        let probe_side_eq_types = self
+            .table_desc
+            .order_key
+            .iter()
+            .map(|order| {
+                self.probe_side_schema.fields[order.index as usize]
+                    .data_type
+                    .clone()
+            })
+            .collect_vec();
+
+        for i in 0..self.build_side_eq_types.len() {
+            if i >= probe_side_eq_types.len()
+                || self.build_side_eq_types[i] != probe_side_eq_types[i]
+            {
+                return Err(ErrorCode::NotImplemented(
+                    "Lookup Joins where the two sides of an equality predicate have different data types".to_string(),
+                    None.into()
+                ).into());
+            }
+        }
```

Comment:
How about simplifying this using iter's all method? For example: `(0..self.build_side_eq_types.len()).iter().all(|idx| build_types[idx] == self.probe_schema.data_types()[idx])`?

Reply:
It can be simplified a bit but `self.probe_schema.data_types()` can't be used because it won't necessarily be the same as `probe_side_eq_types`.